src/EVM/Expr.hs

```diff
@@ -1605,3 +1608,36 @@ concKeccakOnePass orig@(Keccak (CopySlice (Lit 0) (Lit 0) (Lit 64) orig2@(WriteW
     (64, ConcreteBuf a) -> Lit (keccak' a)
     _ -> orig
 concKeccakOnePass x = x
+
+lhsConstHelper ::Expr a -> State Bool (Expr a)
+lhsConstHelper = go
+  where
+    go :: Expr a -> State Bool (Expr a)
+    go e@(Mul (Lit _) _) = pure e
+    go e@(Add (Lit _) _) = pure e
+    go e@(Min (Lit _) _) = pure e
+    go e@(Max (Lit _) _) = pure e
+    go e@(Eq (Lit _) _)  = pure e
+    go e@(And (Lit _) _) = pure e
+    go e@(Or (Lit _) _)  = pure e
+    go e@(Xor (Lit _) _) = pure e
+    go e@(Mul _ (Lit _)) = put False >> pure e
+    go e@(Add _ (Lit _)) = put False >> pure e
+    go e@(Min _ (Lit _)) = put False >> pure e
+    go e@(Max _ (Lit _)) = put False >> pure e
+    go e@(Eq _ (Lit _))  = put False >> pure e
+    go e@(And _ (Lit _)) = put False >> pure e
+    go e@(Or _ (Lit _))  = put False >> pure e
+    go e@(Xor _ (Lit _)) = put False >> pure e
+    go e = pure e
+
+-- Commutative operators should have the constant on the LHS
+checkLHSConstProp :: Prop -> Bool
+checkLHSConstProp a = execState (mapPropM lhsConstHelper a) True
```

Comment:
I think what you are saying would be more complicated :laughing:  But at the same time I'm not 100% sure what you mean. If you wanna write an example, we can check! Actually... I think the Maybe monad would also work, and it would shortcut on Nothing... do you wanna have a go at that? I mean, speed is not _that_ important, but it'd be kinda cool to shortcut -- once it's False of course it'll be False forever, so the Maybe monad's Nothing would fit this role quite well, probably...

Reply:
I still don't get Monads... I am going to defer to your experience here.